enigma/src/main/java/cuchaz/enigma/newabstraction/EntryChange.java

```diff
@@ -0,0 +1,96 @@
+package cuchaz.enigma.newabstraction;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import cuchaz.enigma.EnigmaProject;
+import cuchaz.enigma.source.DecompiledClassSource;
+import cuchaz.enigma.translation.mapping.AccessModifier;
+import cuchaz.enigma.translation.representation.entry.Entry;
+
+public class EntryChange<E extends Entry<?>> {
```

Comment:
Hmm, can you have a private constructor on a record? Cause I'd really hate to have the constructor public on this

Reply:
rip no. In the canonical constructor you cannot even throw checked exceptions, but you can still have public static factory methods to help

Reply:
now I'd have to decide between cleaner API or cleaner implementation
aaAAAAAAAA java why
tbh since I think it's unlikely that fields will be added to this class I'm inclined to leave it as is and not turn it into a record